analyses/infercnv-consensus-cell-type/template-notebooks/SCPCP000015_explore-infercnv-results.Rmd

```diff
@@ -0,0 +1,503 @@
+      cell_group == "reference" ~ "reference cell",
+      stringr::str_detect(ewing_annotation, "tumor") ~ "tumor cell",
+      .default = ewing_annotation
+    )
+  ) |>
+  dplyr::add_count(celltype_fine, name = "celltype_fine_count") |>
+  dplyr::mutate(
+    celltype_fine_n = glue::glue("{celltype_fine} (n={celltype_fine_count})"),
+    # top 10 only
+    celltype_fine_n = forcats::fct_lump(celltype_fine_n, 10),
+  )
+
+# add n() to "Other" for plot clarity and update factor to frequency order
+current_levels <- levels(query_only_df$celltype_fine_n)
+total_other <- query_only_df |>
+  dplyr::filter(celltype_fine_n == "Other") |>
+  nrow()
+updated_other <- glue::glue("Other (n={total_other})")
+
+query_only_df <- query_only_df |>
+  dplyr::mutate(
+    # update other level to have counts
+    celltype_fine_n = as.character(celltype_fine_n),
+    celltype_fine_n = ifelse(celltype_fine_n == "Other", updated_other, celltype_fine_n),
+    # order based on freq
+    celltype_fine_n = as.factor(celltype_fine_n),
+    celltype_fine_n = forcats::fct_infreq(celltype_fine_n),
+    celltype_fine_n = forcats::fct_relevel(celltype_fine_n, updated_other, after = Inf)
+  ) |>
+  # exclude small clusters
+  dplyr::add_count(subcluster, name = "subcluster_size") |>
+  dplyr::filter(subcluster_size >= 10)
+
+
+# 11 hex colors!
+# adapted from 'Classic 10' from paletteer: https://r-graph-gallery.com/color-palette-finder
+colors <- c("#1F77B4", "#FF7F0E", "#2CA02C", "#D6272F", "#9467BD", "#8C564B", "#E377C2", "#7F7F7F", "#BCBD22", "#17CFFB", "#17CF1F")
+```
+
+
+```{r, fig.width = 12}
+ggplot(query_only_df) +
+  aes(
+    x = forcats::fct_infreq(subcluster),
+    fill = celltype_fine_n
+  ) +
+  geom_bar() +
+  scale_fill_manual(values = colors) +
+  theme(axis.text.x = element_blank())
+```
+
+
+
+Finally, let's also look more specifically at the `reference` clusters, where we might expect some batch effects since cells were taken from several different libraries.
+Do originating libraries cluster together?
+When interpreting this plot, bear in mind that there might also be an association between cell types and libraries, and we would expect cell types to still cluster together.
+
+```{r}
+reference_cells_df <- has_cnv_df |>
+  dplyr::filter(cell_group == "reference")
+
+ggplot(reference_cells_df) +
+  aes(x = subcluster, fill = library_id) +
+  geom_bar() +
+  scale_fill_brewer(palette = "Dark2") +
+  theme(
+    legend.position = "bottom",
+    axis.text.x = element_blank()
+  )
+```
+
+
+## CNV across chromosomes
+
+Now, we'll look at CNVs across chromosomes, specifically considering the `proportion_scaled_cnv_` output from the HMM.
+
+Again, these are the specific CNVs we expect to see in Ewing sarcoma:
+
+```{r}
+validation_df
+```
+
+```{r}
+# First, construct data frame with proportion_scaled_ instead of has_cnv
+prop_cnv_df <- infercnv_df |>
+  dplyr::select(
+    library_id,
+    barcodes,
+    starts_with("proportion_scaled_")
+  ) |>
```

Comment:
```suggestion
```
Same thing as above, you shouldn't need to do this.

Reply:
Now this one for sure, since there's no grouped calculation!